[CIRCLE-32867] Adding contexts field restrictions doc (#5147)
circleci/circleci-docs: jekyll/_cci2/contexts.md

If you move your repository to a new organization, you must also have the context with that unique name set in the new organization.

### Contexts & Environment variables validation

When creating contexts/environment variables, please note the following:

- The context name must be 200 or fewer characters, must contain at least one non-whitespace character, and must not contain leading, trailing or vertical whitespaces.
- An organization may contain no more than 500 contexts.
- The environment variable name must be 300 or fewer characters, must have alpha or `_` as the first character and can *only* contain  alphanumeric or `_` as the remaining characters.

> - An environment variable value must be 32000 or fewer characters and must NOT contain `null`. 

Reviewer: I think we can leave out the bit about `NUL` byte - it's not doc'd by any of our competitors, but it's invalid for any POSIX environment and I don't know of a significant use case for it.
```suggestion
- An environment variable value must have 32K or fewer characters. 
```

**Note**: An empty environment variable is considered valid.

> - A context may contain no more than 100 environment variables.

Reviewer: ~~ Same here, just an alternate wording that I thought might work - totally happy either way.
```suggestion
- Each context is limited to 100 environment variables.
```

environment variable name. The request body should include a `value` key 
containing the plaintext secret as a string.

> ##### Environment Variable validation

Reviewer 2: 3 `#`s right? So it's not included as part of `Secure Environment Variable Creation, Deletion, and Rotation`

Author: this is 5 `#`s because it falls under `#### Creating Environment Variables`

###### Environment variable names

> The environment variable name must be 300 or fewer characters, must have alpha or _ as the first character and can contain alphanumeric or _ as the remaining characters.

Reviewer 2: and can *only* contain

Reviewer 3: Can we also wrap underscores in in `backticks` for easier reading?

###### Environment variable values

> An environment variable value must be 32000 or fewer characters and must NOT contain null. 

Reviewer 3: ```suggestion
An environment variable value must be 32000 or fewer characters and must NOT contain `null`. 
```

> **Note**: Empty is a valid option for environment variable values.

Reviewer 3: ```suggestion
**Note**:  An empty environment variable is  considered valid.
```
